src/Language/ASKEE/Exposure/Interpreter.hs

```diff
@@ -322,6 +327,25 @@ interpretCall fun args =
                Right ds -> pure $ VModelExpr $ EVal $ VDataSeries ds
         _ -> typeError "loadCSV expects a single string argument"
 
+    FJoin ->
+      case args of
+        [VModelExpr (EVal (VModel m1)), VModelExpr (EVal (VModel m2)), VArray vs]
+          | all good vs -> 
```

Comment:
This checks if each of the `vs` are of the form `VArray [VString _, VString _]` and later repeats the same check in `mapify`. I'd prefer using pattern guards so that we only need to perform this check once. Something like:

```hs
    FJoin ->
      case args of
        [VModelExpr (EVal (VModel m1)), VModelExpr (EVal (VModel m2)), VArray vs]
          | Just vs' <- traverse twoElemStrArr_maybe vs
          = pure $ VModel $ modelAsCore (join (Map.fromList vs') (coreAsModel m1) (coreAsModel m2)
          | otherwise
          = ...
        _ -> ...
      where
        twoElemStrArr_maybe (VArray [VString v1, VString v2]) = Just (v1, v2)
        twoElemStrArr_maybe _ = Nothing
```

Reply:
An excellent suggestion, thanks!